ui/app/src/components/PageBuilderPanel/PageBuilderPanel.tsx

```diff
@@ -15,33 +15,56 @@
  */
 
 import * as React from 'react';
-import { Suspense } from 'react';
+import { Suspense, useEffect } from 'react';
 import { useDispatch } from 'react-redux';
-import { updatePageBuilderPanelWidth } from '../../state/actions/preview';
+import { initPageBuilderPanelConfig, updatePageBuilderPanelWidth } from '../../state/actions/preview';
 import LoadingState, { ConditionalLoadingState } from '../SystemStatus/LoadingState';
 import { useSelection } from '../../utils/hooks/useSelection';
 import { useActiveSiteId } from '../../utils/hooks/useActiveSiteId';
 import { useActiveUser } from '../../utils/hooks/useActiveUser';
 import { useSiteUIConfig } from '../../utils/hooks/useSiteUIConfig';
 import { renderWidgets } from '../Widget';
 import ResizeableDrawer from '../../modules/Preview/ResizeableDrawer';
+import { usePreviewState } from '../../utils/hooks/usePreviewState';
+import EmptyState from '../SystemStatus/EmptyState';
+import { FormattedMessage } from 'react-intl';
 
 export function PageBuilderPanel() {
   const dispatch = useDispatch();
   const uiConfig = useSiteUIConfig();
+  const { pageBuilderPanel } = usePreviewState();
   const site = useActiveSiteId();
   const { rolesBySite } = useActiveUser();
   const { pageBuilderPanelWidth: width, editMode, pageBuilderPanelStack } = useSelection((state) => state.preview);
   const onWidthChange = (width) => dispatch(updatePageBuilderPanelWidth({ width }));
+
+  useEffect(() => {
+    if (uiConfig.xml && !pageBuilderPanel.widgets) {
+      dispatch(initPageBuilderPanelConfig({ configXml: uiConfig.xml, references: uiConfig.references }));
+    }
+  }, [uiConfig.xml, uiConfig.references, dispatch, pageBuilderPanel.widgets]);
+
   return (
     <ResizeableDrawer open={editMode} belowToolbar anchor="right" width={width} onWidthChange={onWidthChange}>
       <Suspense fallback={<LoadingState />}>
-        <ConditionalLoadingState isLoading={!Boolean(uiConfig.preview.pageBuilderPanel.widgets)}>
-          {renderWidgets(
-            pageBuilderPanelStack.length
-              ? pageBuilderPanelStack.slice(pageBuilderPanelStack.length - 1)
-              : uiConfig.preview.pageBuilderPanel.widgets,
-            rolesBySite[site]
+        <ConditionalLoadingState isLoading={!Boolean(pageBuilderPanel)}>
+          {pageBuilderPanel.widgets ? (
```

Comment:
How do you differentiate between not loaded yet and configuration file missing? Since loading state (above) will never be triggered, looks like you might see the empty state momentarily while init occurs.